subsys/fs/nvs/nvs.c

```diff
@@ -319,37 +320,30 @@ static int _nvs_prev_ate(struct nvs_fs *fs, u32_t *addr, struct nvs_ate *ate)
 		return rc;
 	}
 
-	while (1) {
-		*addr += ate_size;
-		if (((*addr) & ADDR_OFFS_MASK) != fs->sector_size) {
-			break;
-		}
-		/* last ate in sector, do jump to previous sector */
-		if (((*addr) >> ADDR_SECT_SHIFT) == 0) {
-			*addr += (fs->sector_count << ADDR_SECT_SHIFT);
-		}
-		(*addr) -= (1 << ADDR_SECT_SHIFT);
-		while (1) {
-			*addr -= ate_size;
-			rc = _nvs_flash_cmp_const(fs, *addr, 0xff, sizeof(struct nvs_ate));
-			if (!rc) {
-				break;
-			}
-			rc = _nvs_flash_cmp_const(fs, *addr, 0x00, sizeof(struct nvs_ate));
-			if (!rc) {
-				break;
-			}
-			/* stop at the end of the sector in case it is not
-			   closed properly */
-			if (((*addr) & ADDR_OFFS_MASK) == 0) {
-				break;
-			}
-		}
-		/* stop after stepping through all sectors */
-		if (*addr == fs->ate_wra) {
-			break;
-		}
+	*addr += ate_size;
+	if (((*addr) & ADDR_OFFS_MASK) != (fs->sector_size - ate_size)) {
+		return 0;
+	}
+
+	/* last ate in sector, do jump to previous sector */
+	if (((*addr) >> ADDR_SECT_SHIFT) == 0) {
```

Comment:
more legible is:
```suggestion
	if (((*addr) >> ADDR_SECT_SHIFT) == 0) {
		*addr += (fs->sector_count - 1 << ADDR_SECT_SHIFT);
	} else {
	         (*addr) -= (1 << ADDR_SECT_SHIFT);
	}
```

Reply:
agreed, I will change but ad some extra ()